### DISCUSSION TITLE
How to specify trackingMethod at the app level

### DISCUSSION THREAD

**Discussion by vgiannone3** [argoproj/argo-cd#9788] [Q&A] [answered]
Hello! Love ArgoCD, and have been using it for little over a year now. 

Recently I have been trying to enhance my gitops workflows by leveraging KubeVela with ArgoCD, but I have been having some issues. These issues are well known, and are referenced in KubeVela issues [#2608](https://github.com/kubevela/kubevela/issues/2608) and ArgoCD issues [#5082](https://github.com/argoproj/argo-cd/issues/5082), [#8047](https://github.com/argoproj/argo-cd/discussions/8047), and [#3603](https://github.com/argoproj/argo-cd/issues/3603). 

When reading through these issues, the solution to this problem is apparently in [this proposal](https://github.com/argoproj/argo-cd/blob/master/docs/proposals/application-name-identifier.md) which was implemented in ArgoCD [release 2.2](https://github.com/argoproj/argo-cd/releases/tag/v2.2.0) and documented [here](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_tracking/).

But, the documentation only specifies how to set this value within argocd-cm which applies it system wide. I manage many other applications which do not use KubeVela, so applying it system wide will force me to make changes that I do not need to make. 

So, the question is, how do I set trackingMethod at the app level? Once I have this answer, I would be happy to create a pull request with updates to the documentation so it is included there, to help someone who may be having the same problem.

**Comment by jannfis** [comment, marked as answer]
Hi @vgiannone3,

I'm afraid that the application-level setting is not yet implemented, and thus, is also not documented. So your only option for now is to set the tracking method globally.

Be aware of a current bugs before doing that, tho: #9781

**Reply by vgiannone3** [reply]
Thanks for the reply, @jannfis!! Well for now, I may need to go about a workaround for this. Not sure what it will be, but will need to mull over that. 

Are you aware of anyone (or even yourself :)) That has gotten KubeVela OAM type templates successfully deployed leveraging ArgoCD? Specifically, I would like to leverage KubeVela CRDs in my kustomize templates to define components/traits, and use overlays to modify the properties of these components/traits based on an environment. Then, I would like to leverage ArgoCD, and ArgoCD Image Updater for the deployments. 

Honestly, this does not seem like too much of a stretch. OAM fits the GitOps model so well. There are just several key problems that are keeping from realizing this:
1. Resources created by KubeVela do not have an owner reference. So I need to disable pruning which leaves my applications in an OutOfSync state. 
2. Kustomize configMapGenerator uniquely generated configmaps are not updated in the resources created by KubeVela preventing updates from ArgoCD being changed in the live resources. 
4. Image name/tags are not updated using argocd image updater.

This does work today with components like Bitnami secrets, using the owner property, which allows those resources to have that lineage, and other common kubernetes resources like services and endpoint slices. 

Based on my research within this GitHub repo, we are so close to having this working!

**Reply by jannfis** [reply]
Hi @vgiannone3, unfortunately there's another bug that would affect your use-case 1 for using the tracking annotation instead of the label, if I understood you correctly. This bug is documented here: #8683

I don't know too much about KubeVela - in fact, I know close to nothing - to comment about 2 or 3.

**Reply by vgiannone3** [reply]
@jannfis Understood! I will continue hammering away at it. If I figure something out, I will be sure to report it. Thanks!
